src/consumer.ts

```diff
@@ -287,97 +336,51 @@ export class Consumer extends EventEmitter {
     if (this.pollingStartedInstrumentCallback) {
       this.pollingStartedInstrumentCallback({
         instanceId: process.env.HOSTNAME,
-        queueUrl: this.queueUrl
+        queueUrl: this.queueUrl,
+        // instrument i am about to request this.batchSize messages
+        batchSize: this.batchSize
       });
     }
-    const receiveParams = {
-      QueueUrl: this.queueUrl,
-      AttributeNames: this.attributeNames,
-      MessageAttributeNames: this.messageAttributeNames,
-      MaxNumberOfMessages: this.batchSize,
-      WaitTimeSeconds: this.waitTimeSeconds,
-      VisibilityTimeout: this.visibilityTimeout
-    };
 
     let currentPollingTimeout = this.pollingWaitTimeMs;
-    this.receiveMessage(receiveParams)
-      .then(this.handleSqsResponse)
-      .catch(err => {
-        this.emit('error', err);
-        if (isConnectionError(err)) {
-          debug('There was an authentication error. Pausing before retrying.');
-          currentPollingTimeout = this.authenticationErrorTimeout;
-        }
-        return;
-      })
-      .then(() => {
-        setTimeout(this.poll, currentPollingTimeout);
-      })
-      .catch(err => {
-        this.emit('error', err);
-      });
+
+    if (this.batchSize > 0) {
+      const receiveParams = {
+        QueueUrl: this.queueUrl,
+        AttributeNames: this.attributeNames,
+        MessageAttributeNames: this.messageAttributeNames,
+        MaxNumberOfMessages: this.batchSize,
+        WaitTimeSeconds: this.waitTimeSeconds,
+        VisibilityTimeout: this.visibilityTimeout
+      };
+
+      this.receiveMessage(receiveParams)
+        .then(this.handleSqsResponse)
+        .catch(err => {
+          this.emit('error', err);
+          if (isConnectionError(err)) {
+            debug('There was an authentication error. Pausing before retrying.');
+            currentPollingTimeout = this.authenticationErrorTimeout;
+          }
+          return;
+        })
+        .then(() => {
+          setTimeout(this.poll, currentPollingTimeout);
+        })
+        .catch(err => {
+          this.emit('error', err);
+        });
+    } else {
+      setTimeout(this.poll, currentPollingTimeout);
```

Comment:
I see that at you didn't distinguish between the different timeouts (finished polling or didn't poll), but I see that you did pass the variable in the automations PR

Reply:
it is in the 2nd PR :) 
https://github.com/DaPulse/sqs-consumer/pull/2/files